src/Geocoder/Provider/Mapzen.php

```diff
@@ -0,0 +1,223 @@
+<?php
+
+/**
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+namespace Geocoder\Provider;
+
+use Geocoder\Exception\InvalidCredentials;
+use Geocoder\Exception\QuotaExceeded;
+use Geocoder\Exception\NoResult;
+use Geocoder\Exception\UnsupportedOperation;
+use Http\Client\HttpClient;
+
+/**
+ * @author Gary Gale <gary@vicchi.org>
+ */
+class Mapzen extends AbstractHttpProvider
```

Comment:
I've just opened a new issue (#537) so we track adding finalisation to all providers. But I'm also assuming that based on @willdurand's comment above, there's nothing stopping this PR from being merged now?